x/inter-tx/ibc_module.go

```diff
@@ -123,14 +124,16 @@ func (im IBCModule) OnAcknowledgementPacket(
 		return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-27 packet acknowledgement: %v", err)
 	}
 
-	txMsgData := &sdk.TxMsgData{}
-	if err := proto.Unmarshal(ack.GetResult(), txMsgData); err != nil {
+	var txMsgData sdk.TxMsgData
+	if err := proto.Unmarshal(ack.GetResult(), &txMsgData); err != nil {
 		return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-27 tx message data: %v", err)
 	}
 
 	switch len(txMsgData.Data) {
 	case 0:
-		// TODO: handle for sdk 0.46.x
+		for _, msgResp := range txMsgData.GetMsgResponses() {
+			im.keeper.Logger(ctx).Info("msg response in ICS-27 packet", "response", msgResp.GoString(), "typeURL", msgResp.GetTypeUrl())
+		}
```

Comment:
Note, one could handle the msg response using a type switch like so:

```go

switch msg := msgResp.GetCachedValue().(type) {
case *banktypes.MsgSendResponse:
	im.keeper.Logger(ctx).Info("handling bank send msg response in ICS-27 packet", "response", msg)
case *stakingtypes.MsgDelegateResponse:
	im.keeper.Logger(ctx).Info("handling staking delegate msg response in ICS-27 packet", "response", msg)
case nil:
	im.keeper.Logger(ctx).Info("msg response is nil")
}
```

However, because the response values don't actually have any meaningful data, i.e. the return types are usually just something like below [(see here)](https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/keeper/msg_server.go#L266):

```go
return &MsgDelegateResponse{}, nil
```

Then the empty struct will be unmarshalled as `nil` and all we see is the `nil` case being output in the logs.
For example:

```
{"level":"info","module":"x/intertx","time":"2022-08-22T15:29:11+02:00","message":"msg response is nil"}
```